lib/c-hyper.c

```diff
@@ -73,8 +73,6 @@ size_t Curl_hyper_recv(void *userp, hyper_context *ctx,
 
   DEBUGF(infof(data, "Curl_hyper_recv(%zu)", buflen));
   result = Curl_read(data, conn->sockfd, (char *)buf, buflen, &nread);
-  if(!result && !nread)
-    result = CURLE_AGAIN;
```

Comment:
Does there need to be a similar revert for the same logic that was added in Curl_write? I notice #11203 was reopened because having this logic does not fix what was reported.